### PR TITLE
useSpy

### PR DESCRIPTION
**Pull request by JoviDeCroock**
@JonathanCa97 I really don't want to diminish the work you did but I've made a small implementation reusing your code how are you liking this?

Also I've published 3.2.0 with the setter on context.

Size: 1.94 --> 1.98KB

Will still have to add some tests to make codecov happy

### CODE REVIEW

**Comment by JonathanCa97** [comment]
@JoviDeCroock looks good to me! I can see how I could change my current implementation with useSpy. Thx for taking my use case in consideration.

**Comment by JoviDeCroock** [comment]
> @JoviDeCroock looks good to me! I can see how I could change my current implementation with useSpy. Thx for taking my use case in consideration.

Note that 3.2.0 uses useFormConnect because useSpy isn't possible with that emitter

File: __tests__/hooks/useSpy.test.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      const ageField = getByTestId('age');
+      act(() => {
+        fireEvent.change(ageField, {target: {value: 9}})
+      });
+
+      expect(spy).toBeCalledTimes(1);
+      expect(spy).toBeCalledWith("9")
+
+      act(() => {
+        fireEvent.change(ageField, {target: {value: 9}})
```
Comment (JoviDeCroock) [inline __tests__/hooks/useSpy.test.tsx]: Double check it doesn't call it for the same value again

File: src/Form.tsx [hunk 2]
```diff
@@ -161,14 +162,19 @@ const Form = <Values extends object>({
...
       value={{
-        errors: formErrors as Errors,
+        errors: errors as Errors,
         formError,
         isDirty: isDirty.current,
         isSubmitting,
         resetForm,
-        setFieldTouched: (fieldId: string, value?: boolean) => {
+        // TODO: add test
+        setFieldError: (fieldId: string, error?: any) => {
```
Comment (JoviDeCroock) [inline src/Form.tsx]: We need a new method to set a field level error, we provide this in context to help `useFormConnect` and `useSpy`

File: src/useSpy.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { get } from './helpers/operations';
+import { FormHookContext } from './types';
+import { useContextEmitter } from './useContextEmitter';
+
+const useSpy = (fieldId: string, cb: (newValue: any, ctx: FormHookContext) => void) => {
+  const isMounted = React.useRef(false);
+  const ctx = useContextEmitter(fieldId);
+  React.useEffect(() => {
+    if (isMounted.current) cb(get(ctx.values, fieldId), ctx);
```
Comment (JoviDeCroock) [inline src/useSpy.ts]: We only dispatch after initial mount

File: src/useSpy.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { get } from './helpers/operations';
+import { FormHookContext } from './types';
+import { useContextEmitter } from './useContextEmitter';
+
+const useSpy = (fieldId: string, cb: (newValue: any, ctx: FormHookContext) => void) => {
+  const isMounted = React.useRef(false);
+  const ctx = useContextEmitter(fieldId);
```
Comment (JoviDeCroock) [inline src/useSpy.ts]: We ensure context values are updated when the value is

**Pull request merged**